homeassistant/components/yolink/switch.py

```diff
@@ -118,8 +174,16 @@ def update_entity_state(self, state: dict[str, Any]) -> None:
 
     async def call_state_change(self, state: str) -> None:
         """Call setState api to change switch state."""
-        await self.call_device_api("setState", {"state": state})
-        self._attr_is_on = self.entity_description.value(state)
+        params: dict[str, str | int] = {"state": state}
+        if self.entity_description.device_type == ATTR_DEVICE_MULTI_OUTLET:
+            params["chs"] = 1 << self.entity_description.plug_index
```

Comment:
Is the params struct documented somewhere? If it isn't please comment the code to explain what's going on

Reply:
We have two types of outlet, one with only one output and the other with multiple outputs, There will be some differences in the API
outlet device: http://doc.yosmart.com/docs/yolinkapi/Outlet
multi-outlet: http://doc.yosmart.com/docs/yolinkapi/MultiOutlet

Reply:
This is too much protocol details in the integration. The library should abstract the protocol details into a simpler interface of objects and methods. We shouldn't need to pass dicts with specific strings and do bitwise operations to get api values.